location-datastore-rdbms/src/main/java/com/telecominfraproject/wlan/location/datastore/rdbms/LocationDAO.java

```diff
@@ -357,6 +365,26 @@ public Location getTopLevelLocation(long locationId) {
 
         return ret;
     }
+    
+    public List<Location> getAllAncestors(long locationId) {
+        LOG.debug("Looking up all ancestor Locations for child {}", locationId);
+        
+        Location location;
+        try {
+            location = this.jdbcTemplate.queryForObject(SQL_GET_BY_ID,
```

Comment:
why this extra lookup? you can still remove the current location from the ret by the incoming locationId